index.ts

```diff
@@ -166,7 +166,20 @@ interface Card {
 }
 
 let rooms: Room[] = [];
-wss.on("request", function (request) {
+wss.on("request", async function (request) {
+  let name: string, id: string;
+  try {
+    let expressReq = request.httpRequest as express.Request;
+    ({name, id} = await new Promise<{name: string, id: string}>((resolve, reject) => {
```

Comment:
just a note: for improved readability you can omit the generic type for `Promise` since typescript can easily infer it from use and the developer, too, from the assignment:

```suggestion
    ({name, id} = await new Promise((resolve, reject) => {
```